### PR TITLE
Staging Sites: Rename and unify staging selector

### PR DESCRIPTION
**Pull request by katinthehatsite**
* Fix the reverted PR: https://github.com/Automattic/wp-calypso/pull/74835

## Proposed Changes

* We replace the usage of older selectors `client/state/selectors/is-staging-site.js` and `client/state/selectors/is-site-staging.ts` with a new selector `isSiteWpcomStaging`.

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

### CODE REVIEW

File: client/my-sites/site-settings/form-general.jsx [hunk 1]
```diff
@@ -889,7 +889,7 @@ const connectComponent = connect( ( state ) => {
 		isPaidPlan: isCurrentPlanPaid( state, siteId ),
 		isUnlaunchedSite: isUnlaunchedSite( state, siteId ),
 		isWPForTeamsSite: isSiteWPForTeams( state, siteId ),
-		isStagingSite: isSiteStaging( state, siteId ),
+		isWpcomStagingSite: isSiteWpcomStaging( state, siteId ),
```
Comment (wojtekn) [inline client/my-sites/site-settings/form-general.jsx:892]: Would it make sense to use `isSiteWpcomStaging` as a variable name everywhere? Now we have `isWpcomStagingSite` and `isWpcomStaging`, but the selector name is `isSiteWpcomStaging`.

For other selectors, we usually use the variable name the same as the selector name.

**Review by wojtekn — APPROVED** [review]
It works as expected, I left one note related to a consistent naming convention.

File: client/my-sites/site-settings/form-general.jsx (hunk 1 above)
Comment (sejas) [reply client/my-sites/site-settings/form-general.jsx:892]: We are following the pattern that `isWPForTeamsSite`, `isSiteWpcomAtomic` use.
The benefit is that we can destruct the prop without conflicting with the selector name.

Comment (wojtekn) [reply client/my-sites/site-settings/form-general.jsx:892]: Thanks for the explanation. It makes sense just to unify `isWpcomStagingSite` and `isWpcomStaging` then.

**Review by sejas — APPROVED** [review]
Looks solid and updated ! 😄 👍

File: client/my-sites/site-settings/form-general.jsx (hunk 1 above)
Comment (sejas) [reply client/my-sites/site-settings/form-general.jsx:892]: @wojtekn , Do you refer to https://github.com/Automattic/wp-calypso/blob/0ffa3da237dd6ee9fb8c8ba9f578b5018b0e1ac2/client/sites-dashboard/components/sites-site-plan.tsx#L34 ?
Yes, I can rename it to `isWpcomStagingSite` for consistency 👍 .

Comment (sejas) [reply client/my-sites/site-settings/form-general.jsx:892]: Renamed: https://github.com/Automattic/wp-calypso/pull/74889/commits/cb4781d1d2427d3e974f978df79252cd4245e680

**Pull request merged**